src/main/java/use_case/related_words/related_words_generate/RelatedInteractor.java

```diff
@@ -1,22 +1,30 @@
 package use_case.related_words.related_words_generate;
 
+import entity.Word;
 import entity.factories.OriginalRelatedWordFactory;
+import entity.related_words.RelatedWordsSelectionStrategy;
+
+import java.util.List;
 
 public class RelatedInteractor implements RelatedInputBoundary {
     final RelatedWordDataAccessInterface relatedDataAccessObject;
     final RelatedOutputBoundary relatedPresenter;
-    final OriginalRelatedWordFactory relatedWordFactory;
+    final RelatedWordsSelectionStrategy strategy;
+    final OriginalRelatedWordFactory relatedWordsFactory;
```

Comment:
You added a word factory in the interactor which was an extremely smart move. Using word factory classes and interfaces rather than creating words in the file make the code much easier to work off of and extend for the future.